doc/common/Cadl-Project-Scripts.md

```diff
@@ -0,0 +1,127 @@
+# Integrating with cadl sync and generate scripts
+
+There are 2 common scripts provided for each language to be able to generate from within the language
+repo and use the remote cadl definition in the spec repo.
+
+## Prerequisites
+
+There are 4 things that these two scripts expect are set up in your language repo before they will run correctly.
+
+### cadl-location.yaml
+
+This file should live under the project directory for each service and has the following properties
```

Comment:
We have support both remote git url  and local swagger directory.
If user provides (repo, commit), we will use remote git url, if user provides `spec-root-dir`, we will use local swagger directory.

Shall we  add the document about the `spec-root-dir` property to describe the scenario to use local cloned swagger?

Reply:
why would swagger be involved in the cadl scripts?

Reply:
There are two ways to refer to the cadl files. One is a remote git url, the other is the local file path. In the cadl-location.yaml, you can provide (repo, commit) to indicate an remote URL of the cadl, then cadl-project-sync.ps1 will sparse-checkout the cadl folder. Also you can provide a local swagger repo folder via `spec-root-dir`, cadl-project-sync.ps1 just copy cadl filed to the `TempCadlFiles`.

azure-rest-api-specs CI pipeline now uses the second approach.

Reply:
I agree with @chunyu3 we should have a way to run your script without cloning, assuming the clone is already done and trust people that the state of the clone is the right SHA and the right branch, etc. (we discussed this already). Though I'm not sure if this is adressed somewhere else in this document, we're just line 12 :)

Reply:
We do have a way to run the scripts without cloning.  You just put some files in TempCadlFiles and skip the sync script.